src/pages/trader-tools/margin-calculator/_margin-calculator.tsx

```diff
@@ -70,48 +71,26 @@ import {
 } from 'components/elements'
 import Input from 'components/form/input'
 import RightArrow from 'images/svg/tools/black-right-arrow.svg'
-import { useDerivApi } from 'components/hooks/use-deriv-api'
+import useWS from 'components/hooks/useWS'
 
 const MarginCalculator = () => {
+    const { data, is_loading, send } = useWS('active_symbols')
     const [tab, setTab] = useState('Synthetic')
-    const [activeSymbols, setActiveSymbols] = useState([])
-    const [disableDropdown, setDisableDropdown] = useState(true)
-    const [symbolSpotPrice, setSymbolSpotPrice] = useState({})
 
     const onTabClick = (t) => {
         setTab(t)
     }
-    const deriv_api = useDerivApi()
 
     useEffect(() => {
-        const { send } = deriv_api
-        send({ active_symbols: 'full' }, (response: ActiveSymbolsResponse) => {
-            if (!response.error && response.active_symbols.length > 0) {
-                const data = response.active_symbols
-                setActiveSymbols(data)
-                setDisableDropdown(false)
-            }
-        })
-    }, [])
+        send({ active_symbols: 'full' })
+    }, [send])
 
-    useEffect(() => {
-        const tempSpotPrice = {}
-
-        if (activeSymbols.length < 1) {
-            return
-        }
-        activeSymbols.forEach((item) => {
-            tempSpotPrice[item.symbol] = item.spot
-        })
-
-        setSymbolSpotPrice(tempSpotPrice)
-    }, [activeSymbols])
     const fetchTickData = useCallback(
         (selectedSymbol, setAssetPrice) => {
-            const price = symbolSpotPrice[selectedSymbol]
-            setAssetPrice('assetPrice', price)
+            const selected = data.find((item) => item.symbol === selectedSymbol)
+            setAssetPrice('assetPrice', selected.spot)
```

Comment:
For the issue below, the dorpdown is not closing once the item is selected
` Cannot read properties of undefined (reading 'spot')`

![Screenshot 2022-12-21 at 12 07 17 PM](https://user-images.githubusercontent.com/88178645/208833593-18f48f27-d74a-4291-8cb0-fc5c865a34ea.png)